src/OpenApi/src/Extensions/JsonTypeInfoExtensions.cs

```diff
@@ -98,23 +122,14 @@ internal static string GetSchemaReferenceId(this Type type, JsonSerializerOption
             return $"{typeName}Of{propertyNames}";
         }
 
+        // Special handling for generic types that are collections
+        // Generic types become a concatenation of the generic type name and the type arguments
         if (type.IsGenericType)
         {
-            // Nullable types are suffixed with `?` (e.g. `Todo?`)
-            if (type.GetGenericTypeDefinition() == typeof(Nullable<>)
-                && Nullable.GetUnderlyingType(type) is { } underlyingType)
-            {
-                return $"{underlyingType.GetSchemaReferenceId(options)}?";
-            }
-            // Special handling for generic types that are collections
-            // Generic types become a concatenation of the generic type name and the type arguments
-            else
-            {
-                var genericTypeName = type.Name[..type.Name.LastIndexOf('`')];
-                var genericArguments = type.GetGenericArguments();
-                var argumentNames = string.Join("And", genericArguments.Select(arg => arg.GetSchemaReferenceId(options)));
-                return $"{genericTypeName}Of{argumentNames}";
-            }
+            var genericTypeName = type.Name[..type.Name.LastIndexOf('`')];
+            var genericArguments = type.GetGenericArguments();
+            var argumentNames = string.Join("And", genericArguments.Select(arg => arg.GetSchemaReferenceId(options)));
```

Comment:
Is `Null` ok in the name here?

Reply:
This calls into the overload that never exempts primitives so we're covered here. `JsonTypeInfoExtensionsTests.GetSchemaReferenceId_Works` has test cases to cover generics/enumerables with primitive types so we cover this case.